components/ActivityForm/index.js

```diff
@@ -0,0 +1,145 @@
+import styled from "styled-components";
+import { useRouter } from "next/router";
+
+const StyledFormFieldset = styled.fieldset`
+  display: flex;
+  flex-direction: column;
+  gap: 15px;
+  margin: 15px;
+  margin: 20px 30px 50px;
+`;
+const StyledFieldset = styled.fieldset`
+  display: flex;
+  justify-content: space-around;
+  flex-wrap: wrap;
+  margin: 0px;
+  padding: 30px;
+`;
+
+const StyledInputs = styled.input`
+  padding: 10px;
+  margin: 10px;
+`;
+
+const StyledTextarea = styled.textarea`
+  padding: 10px;
+  margin: 10px;
+`;
+export default function ActivityForm({ onSubmit }) {
+  const router = useRouter();
+  function handleSubmit(event) {
+    event.preventDefault();
+    const formElements = new FormData(event.target);
+    const newActivity = Object.fromEntries(formElements);
+    newActivity.categoryIds = formElements.getAll("categoryIds");
+
+    if (newActivity.categoryIds.length === 0) {
+      alert("Please select at least one category.. ");
+      return false;
+    }
+    onSubmit(newActivity);
+    router.push("/");
+  }
+
+  return (
+    <>
+      <form onSubmit={handleSubmit}>
+        <StyledFormFieldset>
+          <legend>Add new Activity</legend>
+          <label htmlFor="title">Title: </label>
+          <StyledInputs
+            id="title"
+            name="title"
+            type="text"
+            placeholder="Please choose title.."
+            maxLength="30"
+            required
+          ></StyledInputs>
+          <label htmlFor="image">ImageUrl: </label>
```

Comment:
```suggestion
          <label htmlFor="imageUrl">ImageUrl: </label>
```
to connect the label to an element use the `id`.